WordPress/gutenberg
# Experiment - try adding a shortcut for hiding the block toolbar and outline

Helps with #40382, #38409
Related - #46192

## What? Why?
A lot of users generally prefer using the floating toolbar, but a long standing issue is that it can get in the way. It can make it harder to select the previous block, and also obscure your view when you're trying to make a design perfect.

This PR implements a new shortcut I called 'peek'. Hold down the 'alt' ('option' on a mac) key to briefly hide the block toolbar, and release again to bring it back.

I'm not sure if 'alt' is the perfect shortcut, but it's easy to test. Does using such an obvious key like this have any implications? While typing, the toolbar is already hidden, so I don't think that's an issue but maybe there are other things I haven't considered.

At the moment I'm looking for feedback on this - is it something that might be useful?

## How?
It required some work, but nothing too major:
- Add support for modifier only shortcuts
- Add support for `keyUp` events in `useShortcut`
- Make `hideBlockInterface` also hide the block outline (this was more difficult than expected!)
- Fix the `hideBlockInterface` action (the `useVisualizer` hook was immediately calling `showBlockInterface` whenever `hideBlockInterface` was called)
- Implement the shortcut

## Testing Instructions
1. Open the site editor (at the moment it only works in the site editor)
2. Select a block
3. Press `alt` or `option` on a mac

Observe that the block toolbar and outline are hidden

## Screenshots or screencast 
Here's an example of using the feature to select two adjacent blocks that would otherwise be obscured by the toolbar:

https://user-images.githubusercontent.com/677833/229679529-7e087321-c988-4987-8b3f-760061b6e0ba.mp4
